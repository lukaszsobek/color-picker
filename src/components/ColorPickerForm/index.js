import React from "react";

const ColorPickerForm = props => {
    const {onChange, onSubmit} = props;
    return (
        <form onSubmit={onSubmit}>
            <input
                name="form_input"
                onChange={onChange}
                autoComplete="off"
            />
            <button>Accept</button>
        </form>
    );
}

export default ColorPickerForm;