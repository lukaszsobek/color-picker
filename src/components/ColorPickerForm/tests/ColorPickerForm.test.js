import ColorPickerForm from "../";

describe("ColorPickerForm", () => {
    it("returns a working form",() => {

        const props = {
            inputValue: "",
            onChange: "onChange",
            onSubmit: "onSubmit"
        }
    
        expect(ColorPickerForm(props)).toMatchSnapshot("");
    
    });
    
    it("returns a form with undefined values",() => {
    
        const props = "";
    
        expect(ColorPickerForm(props)).toMatchSnapshot("");
    
    });
});
