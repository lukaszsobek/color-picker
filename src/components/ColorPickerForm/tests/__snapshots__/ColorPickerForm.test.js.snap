// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPickerForm returns a form with undefined values 1`] = `
<form
  onSubmit={undefined}
>
  <input
    autoComplete="off"
    name="form_input"
    onChange={undefined}
    value={undefined}
  />
  <button>
    Accept
  </button>
</form>
`;

exports[`ColorPickerForm returns a working form 1`] = `
<form
  onSubmit="onSubmit"
>
  <input
    autoComplete="off"
    name="form_input"
    onChange="onChange"
    value=""
  />
  <button>
    Accept
  </button>
</form>
`;
